vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceOverviewFormatter.java

```diff
@@ -0,0 +1,228 @@
+
+    private static final String ID = "id";
+
+    private static final String IMG = "img";
+
+    private static final String IS_MODULE = "isModule";
+
+    private static final String IS_HIDE_INCOMING_LINKS = "isHideIncomingLinks";
+
+    private static final String IS_UNPUBLISHED = "isUnpublished";
+
+    @Autowired
+    private SpaceRepository spaceRepo;
+
+    @Autowired
+    private ModuleRepository moduleRepo;
+
+    @Autowired
+    private JsonHelper jsonHelper;
+
+    /**
+     * creating ArrayNode for space and module node in the spaceoverview graph
+     * 
+     * @param contextPath   This variable holds the contextpath of the application
+     * @param SpaceNodeList List of spaces
+     * @param nodeArray     This gets populated with space node in the spaceoverview
+     *                      graph
+     * @throws JsonProcessingException
+     */
+    private void constructSpaceNodeArray(String contextPath, Iterable<Space> spaceNodeList, ArrayNode nodeArray)
+            throws JsonProcessingException {
+
+        if (spaceNodeList != null) {
+            StringBuilder linkPathBuilder = new StringBuilder();
+            StringBuilder imagePathBuilder = new StringBuilder();
+            for (Space spaceNode : spaceNodeList) {
+                ObjectNode node = createNodeAndSetName(spaceNode);
+                String id = spaceNode.getId();
+                createSpaceNode(linkPathBuilder, imagePathBuilder, spaceNode, node, contextPath, id);
+                nodeArray.add(node);
+            }
+        }
+    }
+
+    /**
+     * creating ArrayNode for space and module node in the spaceoverview graph
+     * 
+     * @param contextPath    This variable holds the contextpath of the application
+     * @param moduleNodeList List of modules
+     * @param nodeArray      This gets populated with module node in the
+     *                       spaceoverview graph
+     * @throws JsonProcessingException
+     */
+    private void constructModuleNodeArray(String contextPath, Iterable<Module> moduleNodeList, ArrayNode nodeArray)
+            throws JsonProcessingException {
+
+        if (moduleNodeList != null) {
+            StringBuilder linkPathBuilder = new StringBuilder();
+            StringBuilder imagePathBuilder = new StringBuilder();
+            for (Module moduleNode : moduleNodeList) {
+                ObjectNode node = createNodeAndSetName(moduleNode);
+                String id = moduleNode.getId();
+                createModuleNode(linkPathBuilder, imagePathBuilder, node, contextPath, id);
+                nodeArray.add(node);
+            }
+        }
+    }
+
+    /**
+     * This is a common method for both space and module
+     * 
+     * @param nodeval This variable store either space/module node
+     * @return After setting name variable of node it returns the node
+     */
+    private ObjectNode createNodeAndSetName(IVSpaceElement nodeval) {
+
+        ObjectMapper mapper = new ObjectMapper();
```

Comment:
there shouldn't be any references to the json classes; that should be encapsulated in the jsonhelper.

Reply:
or put everything in this class and get rid of the jsonhelper (what is it's responsibility here?).